default/AI/UniverseUtilsAI.py

```diff
@@ -0,0 +1,164 @@
+    potential_supply_distance += max(AIDependencies.supply_by_size.values())
+    potential_supply_distance += max(AIDependencies.species_supply_range_modifier.values())
+    potential_supply_distance += 1  # World tree special
+    # TODO: +1 to consider capturing planets with Elevators
+    # if foAI.foAIstate.aggression >= fo.aggression.aggressive:
+    # potential_supply_distance += 1
+    return potential_supply_distance
+
+
+@cache_by_turn
+def get_all_annexable_system_ids():
+    """Get all systems considered annexable.
+
+    :return: of system IDs
+    :rtype: set
+    """
+    # TODO: distinguish colony-annexable systems and outpost-annexable systems
+    # TODO: Make sure, that update annexable systm got called before
+    annexable_system_ids = set()
+    systems_by_supply_tier = get_systems_by_supply_tier()
+    for jumps in range(0, -1-get_potential_supply_distance(), -1):
+        annexable_system_ids.update(systems_by_supply_tier.get(jumps, []))
+    return annexable_system_ids
+
+
+@cache_by_turn
+def check_supply():
+    """Get fleet suppliable planets.
+
+    :return: suppliable planet IDs
+    :rtype: list
+    """
+    empire = fo.getEmpire()
+    fleet_suppliable_system_ids = empire.fleetSupplyableSystemIDs
+    fleet_suppliable_planet_ids = PlanetUtilsAI.get_planets_in__systems_ids(fleet_suppliable_system_ids)
+    print
+    print "    fleet_suppliable_system_ids: %s" % fleet_suppliable_system_ids
+    print "    fleet_suppliable_planet_ids: %s" % fleet_suppliable_planet_ids
+    print
+    print "-------\nEmpire Obstructed Starlanes:"
+    print list(empire.obstructedStarlanes())
+    return fleet_suppliable_planet_ids
+
+
+@cache_by_turn
+def annexable_systems_old():
+    """Find annexable systems.
+
+    :return: set of system IDs indexed by distance to empire supply lines.
+    :rtype: dict
+    """
+    empire = fo.getEmpire()
+    empire_id = empire.empireID
+    universe = fo.getUniverse()
+    potential_supply_distance = get_potential_supply_distance()
+    annexable_system_ids_old = {0: set(empire.fleetSupplyableSystemIDs)}
+    covered_systems = annexable_system_ids_old[0]
+    for distance in xrange(1, potential_supply_distance+1):
+        candidates = set()
+        for sys_id in annexable_system_ids_old[distance-1]:
+            candidates.update(universe.getImmediateNeighbors(sys_id, empire_id))
+        print "covered: ", covered_systems
+        annexable_system_ids_old[distance] = candidates - covered_systems
+        print "annex %d" % distance, annexable_system_ids_old[distance]
+        covered_systems.update(annexable_system_ids_old[distance])
+    print "First Ring of annexable systems:", ppstring(PlanetUtilsAI.sys_name_ids(annexable_system_ids_old.get(2, [])))
+    print "Second Ring of annexable systems:", ppstring(PlanetUtilsAI.sys_name_ids(annexable_system_ids_old.get(3, [])))
+    print "Third Ring of annexable systems:", ppstring(PlanetUtilsAI.sys_name_ids(annexable_system_ids_old.get(4, [])))
```

Comment:
PyCharm settings -> editor general -> ensure line feed on the end of file on save.